nuget/ValidationContext.cs

```diff
@@ -0,0 +1,243 @@
+        ///     <c>ValidationContext</c> was created with <c>useEmbeddedValidator = false</c>.
+        /// </param>
+        /// <param name="validateResources">When set to <c>false</c>, only JSON description is validated.</param>
+        /// <param name="timeout">Overrides default validation timeout.</param>
+        /// <returns>A <c>ValidationReport</c> instance.</returns>
+        /// <exception cref="ArgumentException">Thrown when either asset or custom glTF-Validator path is incorrect.</exception>
+        /// <exception cref="FileNotFoundException">Thrown when glTF asset or custom glTF-Validator is not found.</exception>
+        /// <exception cref="InvalidOperationException">
+        ///     Thrown when the <c>ValidationContext</c> was created without embedded
+        ///     validator but custom one was not provided.
+        /// </exception>
+        /// <exception cref="TimeoutException">Thrown when validation takes longer than allowed time.</exception>
+        /// <exception cref="ApplicationException">Thrown when the validator encounters unhandled runtime error.</exception>
+        public ValidationReport Validate(string path, string customValidatorPath = "", bool validateResources = true,
+            TimeSpan timeout = default)
+        {
+            return ValidateAsync(path, customValidatorPath, validateResources, timeout).Result;
+        }
+
+        /// <summary>
+        ///     Asynchronously runs glTF-Validator on provided asset and returns deserialized report.
+        /// </summary>
+        /// <param name="path">Path to the glTF file.</param>
+        /// <param name="customValidatorPath">
+        ///     Path to the custom glTF-Validator executable. Must be provided if
+        ///     <c>ValidationContext</c> was created with <c>useEmbeddedValidator = false</c>.
+        /// </param>
+        /// <param name="validateResources">When set to <c>false</c>, only JSON description is validated.</param>
+        /// <param name="timeout">Overrides default validation timeout.</param>
+        /// <returns>A <c>ValueTask</c> completing with <c>ValidationReport</c> instance.</returns>
+        /// <exception cref="ArgumentException">Thrown when either asset or custom glTF-Validator path is incorrect.</exception>
+        /// <exception cref="FileNotFoundException">Thrown when glTF asset or custom glTF-Validator is not found.</exception>
+        /// <exception cref="InvalidOperationException">
+        ///     Thrown when the <c>ValidationContext</c> was created without embedded
+        ///     validator but custom one was not provided.
+        /// </exception>
+        /// <exception cref="TimeoutException">Thrown when validation takes longer than allowed time.</exception>
+        /// <exception cref="ApplicationException">Thrown when the validator encounters unhandled runtime error.</exception>
+        public async ValueTask<ValidationReport> ValidateAsync(string path, string customValidatorPath = "",
```

Comment:
I'm not sure why ValueTask is returned here. It probably makes more sense to return Task<ValidationReport> and call Wait on it in the synchronous version.